File: src/main/java/me/zeroeightsix/kami/event/events/RenderPlayerNametagEvent.java
package me.zeroeightsix.kami.event.events;

import me.zeroeightsix.kami.event.KamiEvent;
import net.minecraft.client.network.AbstractClientPlayerEntity;

//TODO: Generify into RenderTagEvent
public class RenderPlayerNametagEvent extends KamiEvent {

    final AbstractClientPlayerEntity entity;

    public RenderPlayerNametagEvent(AbstractClientPlayerEntity entity) {
        this.entity = entity;
    }

    public AbstractClientPlayerEntity getEntity() {
        return entity;
    }

}

File: src/main/kotlin/me/zeroeightsix/kami/event/events/RenderPlayerNametagEvent.kt
package me.zeroeightsix.kami.event.events

import me.zeroeightsix.kami.event.KamiEvent
import net.minecraft.client.network.AbstractClientPlayerEntity

//TODO: Generify into RenderTagEvent
class RenderPlayerNametagEvent(val entity: AbstractClientPlayerEntity) : KamiEvent()